src/clp_ffi_js/ir/StreamReaderDataContext.hpp

```diff
@@ -5,28 +5,26 @@
 #include <utility>
 
 #include <clp/Array.hpp>
-#include <clp/ir/LogEventDeserializer.hpp>
-#include <clp/ir/types.hpp>
-#include <clp/streaming_compression/zstd/Decompressor.hpp>
+#include <clp/ReaderInterface.hpp>
 
 namespace clp_ffi_js::ir {
 /**
  * The data context for a `StreamReader`. It encapsulates a chain of the following resources:
- * A `clp::ir::LogEventDeserializer` that reads from a
- * `clp::streaming_compression::zstd::Decompressor`, which in turn reads from a `clp::Array`.
- * @tparam encoded_variable_t Type of encoded variables encoded in the stream.
+ * A CLP deserializer class that reads from a `clp::ReaderInterface`, which in turn reads from a
```

Comment:
My vision was this interface can be implemented for all kinds of CLP formats including IR and Single File Archive CLP, but by then we should probably rename **Stream**ReaderDataContext to avoid ambiguity. At the moment, it makes sense to add the "IR" text for clarify.